### DISCUSSION TITLE
QOpenGLShader::compile error

### DISCUSSION THREAD

**Discussion by Cantoluna** [CGAL/cgal#6106] [Q&A] [answered]
when I running the example "Surface_mesh/draw_surface_mesh",the terminal tell me "Using context 3.3 GL QOpenGLShader::compile(Vertex): 0:3(1): error: `in' qualifier in declaration of `vertex' only valid for function parameters in GLSL 1.10". How can i solve this problem?

**Comment by sloriot** [comment]
What version of CGAL are you using and on what platform?

**Reply by Cantoluna** [reply]
I use CGAL5.3 on Ubuntu16.04

**Comment by sloriot** [comment, marked as answer]
You are probably having an issue that was fixed in release branch by this PR:
https://github.com/CGAL/cgal/pull/5940

You can try to patch yourself the file or use the version [here](https://github.com/CGAL/cgal/blob/5.3.x-branch/GraphicsView/include/CGAL/Basic_shaders.h).

If it is a CGAL version from the system, you can use the branch version [here](https://github.com/CGAL/cgal/tree/5.3.x-branch).
